### PR TITLE
Add test coverage for TextBox

### PR DESCRIPTION
**Pull request by Eudora-Li01**
Related #https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit tests for TextBox to test its properties: AutoCompleteMode and AutoCompleteSource.
- Add renderer unit tests for TextBox to test its methods: DrawTestBox

## Customer Impact

- None

## Regression? 

- No

## Risk

- Minimal

## Test methodology

- Unit tests
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11021)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    [InlineData(TextBoxState.Hot)]
+    [InlineData(TextBoxState.Selected)]
+    [InlineData(TextBoxState.Disabled)]
+    public void TextBoxRenderer_DrawTextBox(TextBoxState state)
+    {
+        using Bitmap bitmap = new(100, 100);
+        using Graphics graphics = Graphics.FromImage(bitmap);
+        Rectangle bounds = new Rectangle(10, 20, 30, 40);
+
+        TextBoxRenderer.DrawTextBox(graphics, bounds, state);
```
Comment (Epica3055) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs:22]: what is our expected result? no exception or what?

**Comment by LeafShi1** [comment]
@lonitra Please help review this PR

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    [WinFormsTheory]
+    [InlineData(TextBoxState.Normal)]
+    [InlineData(TextBoxState.Hot)]
+    [InlineData(TextBoxState.Selected)]
+    [InlineData(TextBoxState.Disabled)]
+    public void TextBoxRenderer_DrawTextBox(TextBoxState state)
+    {
+        using Bitmap bitmap = new(100, 100);
+        using Graphics graphics = Graphics.FromImage(bitmap);
+        Rectangle bounds = new Rectangle(10, 20, 30, 40);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs:20]: Please use target type new.
```suggestion
        Rectangle bounds = new(10, 20, 30, 40);
```

Same with some other places in this file as well.

File: src/System.Windows.Forms/tests/UnitTests/TextBoxTests.cs [hunk 3]
```diff
@@ -257,6 +257,41 @@ public void TextBox_ImeModeBase_GetWithHandle_ReturnsExpected()
...
+    {
+        using SubTextBox control = new()
+        {
+            AutoCompleteSource = value == AutoCompleteSource.ListItems ? AutoCompleteSource.None : value
+        };
+
+        if(value != AutoCompleteSource.ListItems)
+        {
+            control.AutoCompleteSource.Should().Be(value);
+        }
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/TextBoxTests.cs:282]: should the else cast be tested as well for completeness?

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+        TextBoxRenderer.DrawTextBox(graphics, bounds, "text", SystemFonts.DefaultFont, state);
+    }
+
+    [WinFormsTheory]
+    [InlineData(TextBoxState.Normal)]
+    [InlineData(TextBoxState.Hot)]
+    [InlineData(TextBoxState.Selected)]
+    [InlineData(TextBoxState.Disabled)]
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs:43]: This set of values seems to be used often in these tests. Consider making it into a test data set, similar to `DrawTextBox_FlagsAndState_TestData` below.

File: src/System.Windows.Forms/tests/UnitTests/TextBoxTests.cs [hunk 5]
```diff
@@ -257,6 +257,41 @@ public void TextBox_ImeModeBase_GetWithHandle_ReturnsExpected()
...
+    [WinFormsTheory]
+    [EnumData<AutoCompleteSource>]
+    public void TextBox_AutoCompleteSource_Set_GetReturnsExpected(AutoCompleteSource value)
+    {
+        using SubTextBox control = new()
+        {
+            AutoCompleteSource = value == AutoCompleteSource.ListItems ? AutoCompleteSource.None : value
+        };
+
+        if(value != AutoCompleteSource.ListItems)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/TextBoxTests.cs:279]:
```suggestion
        if (value != AutoCompleteSource.ListItems)
```

File: src/System.Windows.Forms/tests/UnitTests/TextBoxTests.cs (hunk 3 above)
Comment (Eudora-Li01) [reply src/System.Windows.Forms/tests/UnitTests/TextBoxTests.cs:282]: > should the else cast be tested as well for completeness?

AutoCompletSource.ListItems is not supported for TextBox.

Comment (Eudora-Li01) [reply src/System.Windows.Forms/tests/UnitTests/TextBoxTests.cs:282]: AutoCompleteSource.ListItems is not supported for TextBox.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs (hunk 1 above)
Comment (LeafShi1) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs:22]: If the expected result is no exception, you can add a try catch here, like
     try
     {
          TextBoxRenderer.DrawTextBox(graphics, bounds, state);
     }
     catch (Exception ex)
     {
         Assert.True(ex is null, $"Expected no exception, but got: {ex.Message}"); 
     }

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    public void TextBoxRenderer_DrawTextBox_OverloadWithTextFontAndTextBounds(TextBoxState state)
+    {
+        using Bitmap bitmap = new(100, 100);
+        using Graphics graphics = Graphics.FromImage(bitmap);
+        Rectangle bounds = new Rectangle(10, 20, 30, 40);
+        Rectangle textBounds = new Rectangle(10, 20, 30, 40);
+
+        TextBoxRenderer.DrawTextBox(graphics, bounds, "text", SystemFonts.DefaultFont, textBounds, state);
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs:51]: Same problem as above, lack of expected results and actual results.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+public class TextBoxRendererTests
+{
+    public static IEnumerable<object[]> DrawTextBox_State_TestData()
+    {
+        yield return new object[] { TextBoxState.Normal };
+        yield return new object[] { TextBoxState.Hot };
+        yield return new object[] { TextBoxState.Selected };
+        yield return new object[] { TextBoxState.Disabled };
+    }
+
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs:19]: Consider using TheoryData ? See https://github.com/dotnet/winforms/issues/11041

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    [WinFormsTheory]
+    [MemberData(nameof(DrawTextBox_State_TestData))]
+    public void TextBoxRenderer_DrawTextBox(TextBoxState tBState)
+    {
+        using Form form = new();
+        using TextBox textbox = new TextBox();
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs:25]: Please use target type new
using TextBox textBox = new();

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+                );
+    }
+
+    [WinFormsTheory]
+    [MemberData(nameof(DrawTextBox_State_TestData))]
+    public void TextBoxRenderer_DrawTextBox_OverloadWithTextAndFont(TextBoxState tBState)
+    {
+        using Form form = new();
+        using TextBox textbox = new TextBox();
+
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs:55]: Please use target type new
using TextBox textBox = new();

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+            );
+    }
+
+    [WinFormsTheory]
+    [MemberData(nameof(DrawTextBox_State_TestData))]
+    public void TextBoxRenderer_DrawTextBox_OverloadWithTextFontAndTextBounds(TextBoxState tBState)
+    {
+        using Form form = new();
+        using TextBox textbox = new TextBox();
+
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TextBoxRendererTests.cs:90]: Please use target type new
using TextBox textBox = new();

**Review by lonitra — APPROVED** [review]
👍

**Pull request merged**